File: python/prime-factors/prime_factors.py
def factors(natural_number):
    result = []
    i = 2
    while i * i <= natural_number:
        if is_prime(i):
            while natural_number % i == 0:
                result.append(i)
                natural_number //= i

        i += 1
    if natural_number != 1:
        result.append(natural_number)
    return result


def is_prime(n):
    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1
    return True

File: cpp/prime-factors/prime_factors.cpp
#include "prime_factors.h"

namespace prime_factors
{
bool is_prime(int n)
{
    for (int i{2}; i * i <= n; ++i)
    {
        if (n % i == 0)
        {
            return false;
        }
    }

    return true;
}

std::vector<int> of(int n)
{
    std::vector<int> result;
    for (int i{2}; i * i <= n; ++i)
    {
        if (is_prime(i))
        {
            while (n % i == 0)
            {
                result.push_back(i);
                n /= i;
            }
        }
    }

    if (n != 1)
    {
        result.push_back(n);
    }

    return result;
}
} // namespace prime_factors
